src/Gui/SoFCColorGradient.cpp

```diff
@@ -110,20 +110,24 @@ void SoFCColorGradient::setMarkerLabel(const SoMFString& label)
 
 void SoFCColorGradient::setViewportSize(const SbVec2s& size)
 {
-    // don't know why the parameter range isn't between [-1,+1]
-    float fRatio = ((float)size[0]) / ((float)size[1]);
+    float fRatio = static_cast<float>(size[0]) / static_cast<float>(size[1]);
     float fMinX =  4.0f, fMaxX = 4.5f;

```

Comment:
Independent of the PR, I  did not understand where the 4.5f comes from It seems it was found by try and error or do you know the reason?

Reply:
OK, I got it now, the bar has 0.5f width. I will document this.